apps/ledger-live-mobile/src/families/evm/EvmFeesStrategy.tsx

```diff
@@ -54,26 +54,53 @@ export default function EvmFeesStrategy({
     }
   }, [bridge, setTransaction, gasOptions, transaction, shouldPrefillEvmGasOptions]);
 
-  const [customStrategy, setCustomStrategy] = useState(getCustomStrategy(transaction));
+  /**
+   * When the user edits the custom fees, we save the related transaction patch
+   * in order to reapply it the next time the user selects the saved custom
+   * strategy entry in the strategy list.
+   * This is done in order to display the correct fees value for the saved custom
+   * entry in the fees strategy list.
+   * We don't need to save the transaction patch for other fees strategies since
+   * the transaction is already updated with the correct fee data retrieved from
+   * the gasTracker in `prepareTransaction`.
+   * cf. (as of 21-09-2023) https://github.com/LedgerHQ/ledger-live/blob/e87c4c2d879fb9322cd2849f2d34c3be9e500a1e/libs/coin-evm/src/prepareTransaction.ts#L190-L202
+   * But since the custom strategy is handled locally by the client, if we don't
+   * save the patch (including updated fee data) and only update the `feesStrategy`
+   * field, the fees value displayed in the fees strategy for the custom entry
+   * will be the one from the last selected strategy (e.g. slow, medium, fast).
+   */
+  const [customStrategyTransactionPatch, setCustomStrategyTransactionPatch] =
+    useState<Partial<Transaction>>();
+  const [customStrategyFees, setCustomStrategyFees] = useState<BigNumber | null>(null);
 
   useEffect(() => {
-    const newCustomStrategy = getCustomStrategy(transaction);
-
-    if (newCustomStrategy) {
-      setCustomStrategy(newCustomStrategy);
+    if (customStrategyTransactionPatch && transaction.feesStrategy === "custom") {
+      const newCustomStrategyFees = getCustomStrategyFees(transaction);
+      setCustomStrategyFees(newCustomStrategyFees);
     }
-  }, [transaction, setCustomStrategy]);
+  }, [transaction, customStrategyTransactionPatch]);
```

Comment:
We could remove the use of this `useState` and `useEffect` here if we go with the patch proposed bellow